### ISSUE TITLE
[HOLD for payment 2023-03-01] Black area generated above keyboard within company information form

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14716]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open any workspace > Reimburse expenses > Connect bank account > Connect Manually >
2. Enter Routing Number: 123123123
3. Enter Account Number: 1111222233331111
4. Click Continue, it will show Company Information form.
5. Fill up the Company information form one by one field, it will generate black area above keyboard in iOS Safari and Chrome as shown in video.

## Expected Result:
Black area should not generated above keyboard.

## Actual Result:
 It will show black area above keyboard as we fill up the form.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] iOS / Safari and Chrome

**Version Number:** 1.2.63-0
**Reproducible in staging?:** y 
**Reproducible in production?:** Could not check
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/215895310-0602011e-c8a8-4629-9538-d1f5b554eaf4.mov

https://user-images.githubusercontent.com/43996225/215895587-d0a9b142-f224-4537-bc78-f679c3377dcd.MP4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @PrashantMangukiya 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1675181285781359

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012f40a9a8f065b869</li>
        <li>Upwork Job ID: 1620853123237179392</li>
        <li>Last Price Increase: 2023-02-01</li>
    </ul>
</details>

**Comment by puneetlath** [comment]
Thanks @ArekChr! Looking forward to your proposed solution.

**Comment by mo-4444** [comment]
Your issue is in this file 

{maxHeight: this.props.windowHeight}]} 

https://github.com/Expensify/App/blob/93854081336980a19b02138a271fead29b37e5d9/src/components/ScreenWrapper/index.js

**Comment by dhairyasenjaliya** [comment]
**Proposal**

- I think this is a regression from PR https://github.com/Expensify/App/pull/14392 & issue https://github.com/Expensify/App/issues/11463 whereby we are adding maxHeight to `<ScreenWrapper>`  to prevent the button from hiding but on Form pages, we dont really have the floating button which creates blank height to prevent that in all form page where every we used `<ScreenWrapper>` we need to set `null` height 

**Solution 1**

- Manually pass `windowHeight={null}` to  all `<ScreenWrapper>` used in all form pages (ReimbursementAccount) (all steps pages) and all pass `null` to pages that are affected (will need to check affected pages as current GH is only for company steps)

**Example**

```diff  
+    <ScreenWrapper windowHeight={null} includeSafeAreaPaddingBottom={false}>  
```

**Solution 2**

- Define new props example: `allowFloatButton` to `<ScreenWrapper>` component by default we should be kept as false and pass `allowFloatButton={true}` on those affected pages which have the bottom button for example this 2 PR we have 2 different pages whereby we should pass true https://github.com/Expensify/App/issues/11463 ,https://github.com/Expensify/App/issues/14161

- We also need to check which other pages we need to have a floating button 

**Result**


https://user-images.githubusercontent.com/47522946/216273872-b12bde12-eeb7-4dd5-9a3a-ccf84a2fa2b3.mp4

**Comment by mountiny** [comment]
Would it require removing the MaxHeight property or is the regression in the upstream repo update?

**Comment by mountiny** [comment]
@rawalyogendra Would you be able to look into this regression fix? 🙇

**Comment by rawalyogendra** [comment]
@mountiny yes, I am looking into this.

> Yeah, removing `MaxHeight` will fix it. or we have to be intelligent while setting its value. When keyboard opens max height will reduce to a small viewport.


@parasharrajat yes, I agree, I was unaware that `ScreenWrapper` was being used in this way. I am looking for a fix. I will update you guys.

CC: @flodnv @mananjadhav keeping you guys in loop here

**Comment by rawalyogendra** [comment]
### Problem:
I have also discussed this a few weeks back here https://github.com/Expensify/App/issues/13491#issuecomment-1397885379.

In WebKit based mobile browsers, whenever there is an input focused on a webpage which below ~50% height of the Viewport, the browsers scroll the webpage even when the page should not scroll. [Relevant Bug](https://bugs.webkit.org/show_bug.cgi?id=141832#c5)

<details>
<summary>Safari Animation</summary>

![1 OkjvY4cvnTRdaY4gtCl8Fw](https://user-images.githubusercontent.com/7548730/213566973-ec1f3398-b15d-4224-8335-8982a2aca22d.gif)

</details>

Recently, Chrome Developers also made some changes to how the [Layout Viewport behaves when the on-screen keyboard (OSK) gets shown](https://developer.chrome.com/blog/viewport-resize-behavior/).  This is causing similar issues in mWeb (Android) as well.

From the Blog Post:
>As of Chrome 108, Chrome on Android will adjust its viewport resize behavior to no longer resize the Layout Viewport when the on-screen keyboard is shown.
>
>This will align the behavior of Chrome on Android with that of Chrome on iOS, iPadOS, Windows, and CrOS, Safari on iOS and iPadOS, and Edge on iOS, iPadOS, and Windows.

So there are 2 main problems here in both Android and iOS (mWeb):
1. The Page is scrollable, even when it should not, causing blank spaces at the bottom.
2. The page is scrolled when input is focused, creating empty spaces due to first reason.

### Solution:
I have pushed my changes here https://github.com/rawalyogendra/App/tree/ios-scroll-fix

**1. mWeb(Android)**
From the Google Developer's blog post:
>To opt back in to the “old” Chrome on Android behavior, set the viewport meta tag to this:
>```html
><meta name="viewport" content="width=device-width, initial-scale=1.0, interactive-widget=resizes-content">
>```
We can add `interactive-widget=resizes-content` in the viewport meta tag which will fix all the issues related to mWeb(Android).
<details>
<summary>mWeb Android Fix</summary>

https://user-images.githubusercontent.com/7548730/217104784-cb88d865-acf8-4a16-a135-b8974fd39d76.mp4

</details>


**2. mWeb(iOS)**
Even if we revert the changes from this PR https://github.com/Expensify/App/pull/14392, these issues still exist in the application. Here is a screen recording after reverting the change.
<details>
<summary>mWeb iOS Issues</summary>

https://user-images.githubusercontent.com/7548730/217105329-084c86f8-324a-4b62-95f8-6a4eb94241f7.mov

</details>

I could not find a straight forward way to fix these issues. So I looked around for other projects where these issues are already addressed. I came across Adobe's `react-spectrum`
https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/overlays/src/usePreventScroll.ts

To fix the auto scrolling issue, they have used a hack where an `<input/>` element is translated to the top of the page when focused, then it is again translated back to its original position.
Inspired by this, I have tried to manipulate the opacity of the `<input/>`.
https://github.com/Expensify/App/blob/5b50a42896a1e4eb901737396af106c4d6ff0fa6/web/index.html#L52-L60

This surprisingly fixes the auto scrolling issue.

I have also disabled `overscroll-behaviour` and `touch-action` in `body` tag, as we do not want the body to scroll.
https://github.com/Expensify/App/blob/5b50a42896a1e4eb901737396af106c4d6ff0fa6/web/index.html#L29-L32

<details>
<summary>mWeb iOS Fix</summary>

https://user-images.githubusercontent.com/7548730/217112698-63078519-057c-4856-8c61-364126ee375c.mov

</details>

This fixes the scrolling issue in some cases however it is still scrollable, if the underlying content is scrollable, which I think is out of scope for this issue.

**_At last, I want to say that the recent changes may have highlighted existing problems in the application, but these issues were not introduced by the changes itself. They were already present beforehand._**

**Comment by flodnv** [comment]
@parasharrajat @mananjadhav what do you think of @rawalyogendra's proposal? It looks good to me. Why don't we move forward with it and take it from there?

**Comment by sobitneupane** [comment]
[The PR](https://github.com/Expensify/App/pull/14392) was reviewed by @mananjadhav. I commented on the PR regarding another issue.

**Comment by rawalyogendra** [comment]
@sobitneupane can you please test my fix from [here](https://github.com/rawalyogendra/App/tree/ios-scroll-fix) on mWeb(Android & iOS).

Maybe I was not clear in my comment above, but I have a solution for this issue.

**Comment by rawalyogendra** [comment]
@parasharrajat I am sorry for the misunderstanding, but I was not asking for any additional payment.

I was just stating the root cause of the problem and how it may not be related to the original PR.

**Comment by MelvinBot** [comment]
📣 @mananjadhav You have been assigned to this job by @puneetlath!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by puneetlath** [comment]
Ok I'm assigning @mananjadhav as the C+ since they were the C+ on the original which caused the regression. @mananjadhav what do you think of the proposal?

**Comment by mananjadhav** [comment]
@rawalyogendra Your changes look fine to me. But do they cover all the regressions? including this [one](https://github.com/Expensify/App/issues/14716) ?


cc - @flodnv @puneetlath

**Comment by rawalyogendra** [comment]
@mananjadhav yes the screen is not scrolled when input is focused

**Comment by flodnv** [comment]
Go for it @rawalyogendra 🙇

**Comment by MelvinBot** [comment]
📣 @rawalyogendra You have been assigned to this job by @flodnv!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by rawalyogendra** [comment]
@flodnv I have opened the PR. You can check it out

**Comment by JmillsExpensify** [comment]
Linked PR was just merged.

**Comment by MelvinBot** [comment]
@puneetlath, @mananjadhav, @rawalyogendra Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by puneetlath** [comment]
PR was merged today.

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.75-0 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15100


If no regressions arise, payment will be issued on 2023-03-01. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
📣 @MelvinBot! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mananjadhav / @puneetlath] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mananjadhav / @puneetlath] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mananjadhav / @puneetlath] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@rawalyogendra] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@puneetlath] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by mananjadhav** [comment]
@rawalyogendra Can you please update with the Regression Test Proposal here?

**Comment by rawalyogendra** [comment]
1. BUG: Black area generated above keyboard within company information form
- Proposed Test Steps:
  -  Open a workspace > Reimburse expenses > Connect bank account > Connect Manually
  - Enter Routing Number: 123123123
  - Enter Account Number: 1111222233331111
  - Click on Continue, it shows Company Information form.
  - Fill up the Company information form.
  - Verified that it did not generate black area above keyboard in iOS Safari and Chrome.
2. BUG: The Close account button is cut off and has extra padding
- Proposed Test Steps:
  - Go to staging.new.expensify.com
  - Login with any account
  - Navigate to Settings > Security
  - Click on "Close Account"
  - Enter the email in the input and open keyboard
- Do we agree :+1: or :-1:

**Comment by MelvinBot** [comment]
📣 @rawalyogendra! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by rawalyogendra** [comment]
Contributor details
Your Expensify account email: yogen.rawal@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~016922834caeca0d87

**Comment by mananjadhav** [comment]
@puneetlath this has the title for payment, but there won't be any payout for this one. Instead it should be #11463.

**Comment by puneetlath** [comment]
Yes, makes sense. Thanks for reminding me. I'm going to go ahead and close this out in favor of that issue.

**Comment by PrashantMangukiya** [comment]
@puneetlath reporting bonus not issued for this. As I reported this issue.

**Comment by PrashantMangukiya** [comment]
Contributor details
Your Expensify account email: pbMangukiya@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01c718889ed7821f82

**Comment by MelvinBot** [comment]
⚠️ Invalid email. Please make sure to create an Expensify account with this email first [here](https://new.expensify.com/).

**Comment by PrashantMangukiya** [comment]
Contributor details
Your Expensify account email: pbMangukiya@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01c718889ed7821f82

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by PrashantMangukiya** [comment]
Thank you @puneetlath Applied for the job.

**Comment by puneetlath** [comment]
Reporting bonus paid. Closing again.

**Comment by PrashantMangukiya** [comment]
Received. Thank you @puneetlath

**Comment by s77rt** [comment]
We are looking for another fix to this issue as the approved proposal caused a regression https://github.com/Expensify/App/issues/16082#issuecomment-1486667089.
